lib/sqlalchemy/orm/mapped_collection.py

```diff
@@ -76,38 +93,52 @@ class _SerializableColumnGetterV2(_PlainColumnGetter):
 
     __slots__ = ("colkeys",)
 
-    def __init__(self, colkeys):
+    def __init__(
+        self, colkeys: List[Tuple[Optional[str], Optional[str]]]
+    ) -> None:
         self.colkeys = colkeys
         self.composite = len(colkeys) > 1
 
-    def __reduce__(self):
+    def __reduce__(
+        self,
+    ) -> Tuple[
+        Type[_SerializableColumnGetterV2],
+        Tuple[List[Tuple[Optional[str], Optional[str]]]],
+    ]:
         return self.__class__, (self.colkeys,)
 
     @classmethod
-    def _reduce_from_cols(cls, cols):
-        def _table_key(c):
+    def _reduce_from_cols(
+        cls, cols: List[ColumnElement[_KT]]
+    ) -> Tuple[
+        Type[_SerializableColumnGetterV2],
+        Tuple[List[Tuple[Optional[str], Optional[str]]]],
+    ]:
+        def _table_key(c: ColumnElement[_KT]) -> Optional[str]:
             if not isinstance(c.table, expression.TableClause):
                 return None
             else:
-                return c.table.key
+                return c.table.key  # type: ignore
```

Comment:
what's the error?

Reply:
There are two strange errors if remove this ignore:

`lib/sqlalchemy/orm/mapped_collection.py:123:` `error: Returning Any from function declared to return "Optional[str]"  [no-any-return]`

`lib/sqlalchemy/orm/mapped_collection.py:123:` `error: "TableClause" has no attribute "key"  [attr-defined]`

I don't think it's typing issues. And I could't find a way to resolve them